gnomad_qc/v4/create_release/create_de_novo_release.py

```diff
@@ -42,343 +31,248 @@
+       - de novos at all confidence levels
+       - filtered high-confidence coding de novos
 
-
-def custom_info_select(ht: hl.Table, **_) -> Dict[str, hl.expr.Expression]:
-    """
-    Select fields for info Hail Table annotation in de novo release.
-
-    :param ht: Info Hail Table.
-    :return: Select expression dict.
+    :param ht: De novo calls Table.
+    :return: Aggregated and annotated Tables.
     """
-    return {"allele_info": ht.allele_info.drop("nonsplit_alleles")}
+    # Get needed annotation Tables.
+    filters_ht = final_filter(all_variants=True, only_filters=True).ht()
+    freq_ht = public_release("exomes").ht()
+    vep_ht = get_vep(data_type="exomes").ht()
 
+    # Filter to variants with a confidence level
+    ht = ht.filter(~hl.is_missing(ht.de_novo_call_info.confidence))
+    logger.info(f"Getting {ht.count()} de novos with a confidence level...")
 
-def custom_an_select(ht: hl.Table, **_) -> Dict[str, hl.expr.Expression]:
-    """
-    Select AN for release.
-
-    :param ht: Hail Table with AN.
-    :return: Select expression dict.
-    """
-    selects = {"exomes_an": ht.AN, **custom_region_flags_select(ht, data_type="")}
-    selects["region_flags"] = selects["region_flags"].annotate(
-        **{r: ht[r] for r in ht.row_value if r.endswith("_region")}
+    ht = filter_low_conf_regions(
+        ht, filter_lcr=True, filter_decoy=False, filter_segdup=True
+    )
+    logger.info(
+        f"Getting {ht.count()} de novos after removing low confidence regions..."
     )
 
-    return selects
-
+    diploid_expr, hemi_x_expr, hemi_y_expr = get_copy_state_by_sex(
+        ht.locus,
+        ht.is_xx,
+    )
 
-def custom_an_select_globals(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
-    """
-    Select AN Table globals for release.
+    adj_trio = (
+        hl.case()
+        .when(
+            diploid_expr,
+            ht.proband_entry.adj & ht.father_entry.adj & ht.mother_entry.adj,
+        )
+        .when(hemi_x_expr, ht.proband_entry.adj & ht.mother_entry.adj)
+        .when(hemi_y_expr, ht.proband_entry.adj & ht.father_entry.adj)
+        .or_missing()
+    )
 
-    :param ht: Hail Table with AN.
-    :return: Select expression dict.
-    """
-    return {
-        "exomes_an_globals": hl.struct(
-            meta=ht.strata_meta,
-            sample_count=ht.strata_sample_count,
+    ht = ht.annotate(adj_trio=adj_trio)
+
+    ht = (
+        ht.group_by("locus", "alleles")
+        .aggregate(
+            de_novo_AC=hl.struct(
+                AC_all=hl.agg.count_where(ht.adj_trio),
+                AC_all_raw=hl.agg.count(),
+                AC_high_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "HIGH") & ht.adj_trio
+                ),
+                AC_medium_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "MEDIUM") & ht.adj_trio
+                ),
+                AC_medium_conf_P_0_9=hl.agg.count_where(
```

Comment:
when we finalize this threshold, can you document how this threshold was chosen and why?